java/core/src/main/java/co/worklytics/psoxy/rules/Rules2.java

```diff
@@ -90,6 +92,17 @@ public static class Endpoint {
 
         String pathRegex;
 
+        //if provided, only query params in this list will be allowed
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        List<String> allowedQueryParams;
```

Comment:
I'd renamed to something like `onlyAllowedQueryParams`; its absence will enforce that everything is allowed; its inclusion will mean that *only* the ones defined are allowed